numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -550,27 +550,21 @@ voidtype_repr(PyObject *self)
             return NULL;
         }
 
-        return PyObject_CallFunction(reprfunc, "O", self);
+        return PyObject_CallFunction(reprfunc, "O", input);
     }
-    return _void_to_hex(s->obval, s->descr->elsize, "void(b'", "\\x", "')");
+    return _void_to_hex(s->obval, s->descr->elsize, pos1, "\\x", pos2);
 }
 
 static PyObject *
-voidtype_str(PyObject *self)
+voidtype_repr(PyObject *self)
 {
-    PyVoidScalarObject *s = (PyVoidScalarObject*) self;
-    if (PyDataType_HASFIELDS(s->descr)) {
-        static PyObject *reprfunc = NULL;
-
-        npy_cache_import("numpy.core.arrayprint",
-                         "_void_scalar_repr", &reprfunc);
-        if (reprfunc == NULL) {
-            return NULL;
-        }
+    return voidtype_repr_handler(self, "void(b'", "')");
+}
 
-        return PyObject_CallFunction(reprfunc, "O", self);
```

Comment:
I'd maybe be inclined just to extract the contents of this `if` statement into a helper function - we might want to have different behavior for `str` and `repr` of a structured void in future.

Reply:
And then
```
static PyObject *
voidtype_str(PyObject *self)
{
    PyVoidScalarObject *s = (PyVoidScalarObject*) self;
    if (PyDataType_HASFIELDS(s->descr)) {
        return _void_scalar_repr(self);
    }
    return _void_to_hex(s->obval, s->descr->elsize, "b'", "\\x", "'");
}
```

I'd prefer for every void member function to have an `if (PyDataType_HASFIELDS ...)` member to reinforce the idea that  structured types and raw-byte types should be two separate things